schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+        if (success) {
+            thing->setStateValue("connected", true);
+        }
+    });
+
+    // Handle property changed signals
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::chargeTimeChanged, this, [thing](quint32 chargeTime){
+        quint32 chargeTimeMinutes{chargeTime / 60};
+        qCDebug(dcSchneiderElectric()) << thing << "charging time changed" << chargeTimeMinutes << "minutes";
+        thing->setStateValue("chargeTime", chargeTimeMinutes);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::cpwStateChanged, this, [thing, this](SchneiderWallboxModbusTcpConnection::CPWState cpwState){
+        qCDebug(dcSchneiderElectric()) << thing << "CPW state changed" << cpwState;
+        setCpwState(thing, cpwState);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::lastChargeStatusChanged, this, [thing, this](SchneiderWallboxModbusTcpConnection::LastChargeStatus lastChargeStatus){
+        qCDebug(dcSchneiderElectric()) << thing << "Last charge status changed" << lastChargeStatus;
+        setLastChargeStatus(thing, lastChargeStatus);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::stationEnergyChanged, this, [thing](quint32 stationEnergyInWattHours){
+        double stationEnergyInKilowattHours{(static_cast<double>(stationEnergyInWattHours)) / 1000.0};
+        qCDebug(dcSchneiderElectric()) << thing << "station energy changed" << stationEnergyInKilowattHours << "kWh";
+        thing->setStateValue("totalEnergyConsumed", stationEnergyInKilowattHours);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::stationPowerTotalChanged, this, [thing](float stationPowerTotalKilowatt){
+        double stationPowerTotalWatt{stationPowerTotalKilowatt * 1000};
+        qCDebug(dcSchneiderElectric()) << thing << "station total power changed" << stationPowerTotalWatt << "W";
+        thing->setStateValue("currentPower", stationPowerTotalWatt);
+        thing->setStateValue("charging", stationPowerTotalWatt > 20);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::errorStatusChanged, this, [thing, this](quint32 errorBits){
+        qCDebug(dcSchneiderElectric()) << thing << "Error state changed" << errorBits ;
+        setErrorMessage(thing, errorBits);
+    });
+
+    schneiderWallboxTcpConnection->connectDevice();
+
+    connect(schneiderWallbox, &SchneiderWallbox::phaseCountChanged, this, [thing, this](quint16 phaseCount){
+        qCDebug(dcSchneiderElectric()) << thing << "Phase count changed" << phaseCount ;
+        setPhaseCount(thing, phaseCount);
+    });
+
+    /* Probably not needed.
+    connect(schneiderWallbox, &SchneiderWallbox::currentPowerChanged, this, [thing, this](double currentPower){
```

Comment:
if the modbus connection offers a register that gives the corrent currentPower value, indeed I would recommend to delete this code and the custom currentPower calculation in the SchneiderWallbox class to have less confusion what's going on when someone looks at this plugin in 6 months from now.